src/sentry/api/endpoints/organization_release_commits.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import absolute_import
+
+from sentry.api.base import DocSection
+from sentry.api.bases.organization import OrganizationReleasesBaseEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.models import Release, ReleaseCommit
+
+
+class OrganizationReleaseCommitsEndpoint(OrganizationReleasesBaseEndpoint):
+    doc_section = DocSection.RELEASES
+
+    def get(self, request, organization, version):
+        """
+        List an Organization Release's Commits
+        ``````````````````````````````````````
+
+        Retrieve a list of commits for a given release.
+
+        :pparam string organization_slug: the slug of the organization the
+                                          release belongs to.
+        :pparam string version: the version identifier of the release.
+        :auth: required
+        """
+        try:
+            release = Release.objects.get(
+                organization_id=organization.id,
+                projects=self.get_allowed_projects(request, organization),
+                version=version,
+            )
+        except Release.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        queryset = ReleaseCommit.objects.filter(
+            release=release,
+        ).select_related('commit', 'commit__author')
+
+        return self.paginate(
+            request=request,
+            queryset=queryset,
+            order_by='order',
+            on_results=lambda x: serialize([rc.commit for rc in x], request.user),
```

Comment:
If we're only serializing the Commit objects instead of the ReleaseCommit, why don't we just query for Commits instead? Seems a bit awkward going through ReleaseCommit just to get that.

You could just do:

```python
Commit.objects.filter(releasecommit__release=release)
```

And you'll directly have the objects you want. Then the `order_by` becomes `releasecommit__order`.

This gives us SQL like:

```sql
SELECT "sentry_commit"."id", "sentry_commit"."organization_id", "sentry_commit"."repository_id", "sentry_commit"."key", "sentry_commit"."date_added", "sentry_commit"."author_id", "sentry_commit"."message"
FROM "sentry_commit"
INNER JOIN "sentry_releasecommit" ON (
  "sentry_commit"."id" = "sentry_releasecommit"."commit_id"
)
WHERE "sentry_releasecommit"."release_id" = 1 
ORDER BY "sentry_releasecommit"."order" ASC
```

Unless I'm missing something here with what you're doing.